ca_cdk_constructs/ecr/ecr_repository.py

```diff
@@ -0,0 +1,92 @@
+from aws_cdk.aws_ecr import Repository, LifecycleRule, TagStatus
+from aws_cdk import Duration, aws_iam as iam, CfnOutput
+from constructs import Construct
+
+
+class ECRRepository(Construct):
+    """
+    An ECR repository with default lifecycle rules.
+
+    The default rules are as follows:
+    - Remove all untagged images after 1 day
+    - Remove all development images after 90 days. Dev image tags start with `dev_`
+    - Limit the max number of images to 1000
+
+    Max development images and max images are configurable.
+
+    :param scope: The scope of the construct, usually self.
+    :param id: The id of the construct.
+    :param name: The name of the repository. Must be unique within the account.
+    :param additional_lifecycle_rules: Additional lifecycle rules to add in addition to the default rules. Priority must be unique and equal to or less than 10 as not to conflict with existing rules.
+    :param additional_accounts_pull: Additional accounts that are allowed to pull from the repository in the form of their account number
+    :param additional_accounts_push: Additional accounts that are allowed to push to the repository in the form of their account number.
+    :param dev_image_max_age: The maximum age of a development image in days. Defaults to 90.
+    :param max_images: The maximum number of images allowed in the repository. Defaults to 1000.
+    :param scan_on_push: Whether to scan for vulnerabilities on the image on push. Defaults to True.
+    """
+
+    def __init__(
+        self,
+        scope: Construct,
+        id: str,
+        name: str,
+        additional_lifecycle_rules: list[LifecycleRule] = [],
+        additional_accounts_pull: list[str] = [],
+        additional_accounts_push: list[str] = [],
+        dev_image_max_age: int = 90,
+        max_images: int = 1000,
+        scan_on_push: bool = True,
+        outputs_enabled: bool = True,
+    ):
+        super().__init__(scope, id)
+        lifecycle_rules = []
+
+        lifecycle_rules.append(
```

Comment:
let's list the rules in priority order (lowest to highest) as this is how they are evaluated